webapp/src/main/java/io/github/microcks/web/SoapController.java

```diff
@@ -134,6 +140,9 @@ public ResponseEntity<?> execute(
       // Enhancement : if not found, try getting operation from soap:body directly!
       if (rOperation == null) {
          String operationName = extractOperationName(body);
+         if (action == null || action.length() > 0) {
+            action = operationName;
+         }
```

Comment:
I don't get this condition purpose... At this point of flow, we haven't found `rOperation` using the `action` so we're looking at operation extracted from the body payload. 
I think we could directly set `action = operationName` no?

Reply:
As the requested action in header takes precedence over the requested action in the body, I return the header one if the client asks for action in both header and body.
I set the action variable only in case the client asks for action in body only

Reply:
But `action == null || action.length() > 0` is only true if action is null or not empty, right? 
So we have 2 cases here:
* client didn't ask for action in header (action is null because not found in header)
* client did ask for action in header (action is not empty as found in header).

Are you sure it's a `||`? I don't see why we couldn't just have `action = operationName`... sorry.

Reply:
Yes you're right, the condition was wrong. It should be :
`action == null || action.length() == 0`

I changed the way to check by using hasText from Spring utils. It is more readable.